byway/src/main/webapp/routepage.js

```diff
@@ -317,10 +325,16 @@ function sendEmail(){
   } else{ // TO DO: Use alerts to notify user
     console.log("Please enter a valid email address.");
   }
+
+}
+
+/** Opens route on Google Maps */
+function openInGMaps(){
+  window.open(generateRouteLink());
```

Comment:
For URLs it's preferable to update an `<a href=>` whenever the link changes rather than set an on-click handler, because onclick doesn't provide the same set of allowed operations:

onclick:
- click button to open however the default `window.open` handler opens it.

href:
- left click to open in this window.
- middle click to open in a new window/tab
- right click for menu options such as "copy link address"
- hover with mouse to see the URL that clicking will take you to.

Reply:
Changed, thanks for explaining the difference I wasn't aware of the special operations href had.

Reply:
If you update the href in an onclick handler, then the other functionality you get from using href does apply if you haven't previously clicked the button, since the href will still be the default value. You would want to update the href whenever the origin, destination, or waypoints change.

Reply:
I know what you're saying, but I've tested it and it seems that the function is fast enough to keep up with the current route and assign it on click.

Reply:
Did you test *all* interactions?

https://jsfiddle.net/rzu2p0va/

Look at this example, with just a link with a click handler. Click "run" before doing each of these tests (to reset the link, since once you click it, the URL will be changed):

- Hover over the link: `example.com`, but if you click it takes you to `google.com` (so the link inaccurately shows the URL it's pointed at)
- Middle click: goes to `example.com`, so the URL is *not* updated. In the context of your trip generator, that would mean the route you see if you *click* the link is different from the route you see if you *middle click* the link.
- Right click, copy URL, paste in new tab: same as middle click; it will copy an old URL to an old route from the *last* time you clicked the button.

Reply:
You're right, I only tested the two main interactions (left and right click), I changed it so the link is updated with changes to the route. Thanks!